contracts/strategies/StrategyUSDT3pool.sol

```diff
@@ -0,0 +1,431 @@
+        }
+    }
+}
+
+interface Controller {
+    function vaults(address) external view returns (address);
+    function strategies(address) external view returns (address);
+    function rewards() external view returns (address);
+    function approveStrategy(address, address) external;
+    function setStrategy(address, address) external;
+    function approvedStrategies(address, address) external view returns (bool);
+    function withdrawAll(address) external;
+}
+
+interface yvERC20 {
+  function deposit(uint) external;
+  function withdraw(uint) external;
+  function earn() external;
+  function getPricePerFullShare() external view returns (uint);
+}
+
+interface ICurveFi {
+    function get_virtual_price() external view returns (uint);
+    function balances(uint) external view returns (uint);
+    function add_liquidity(
+        uint256[3] calldata amounts,
+        uint256 min_mint_amount
+    ) external;
+    function remove_liquidity(
+        uint256 _amount,
+        uint256[3] calldata min_amounts
+    ) external;
+    function remove_liquidity_one_coin(
+        uint256 _token_amount, 
+        int128 i, 
+        uint256 min_amount
+    ) external;
+    function remove_liquidity_imbalance(
+        uint256[3] calldata amounts,
+        uint256 max_burn_amount
+    ) external;
+    function exchange(
+        int128 from, int128 to, uint256 _from_amount, uint256 _min_to_amount
+    ) external;
+}
+
+/*
+
+ A strategy must implement the following calls;
+ 
+ - deposit()
+ - withdraw(address) must exclude any tokens used in the yield - Controller role - withdraw should return to Controller
+ - withdraw(uint) - Controller | Vault role - withdraw should always return to vault
+ - withdrawAll() - Controller | Vault role - withdraw should always return to vault
+ - balanceOf()
+ 
+ Where possible, strategies must remain as immutable as possible, instead of updating variables, we update the contract by linking it in the controller
+ 
+*/
+
+
+contract StrategyUSDT3pool {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address constant public want = address(0xdAC17F958D2ee523a2206206994597C13D831ec7);
+    address constant public _3pool = address(0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7);
+    address constant public _3crv = address(0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490);
+    address constant public y3crv = address(0x9cA85572E6A3EbF24dEDd195623F188735A5179f);
+
+    address public governance;
+    address public controller;
+    address public strategist;
+
+    uint constant public DENOMINATOR = 10000;
+    uint public treasuryFee = 500;
+    uint public withdrawalFee = 0;
+    uint public strategistReward = 50;
+    uint public threshold = 6000;
+    uint public slip = 100;
+    uint public tank = 0;
+    uint public p = 0;
+
+    event Threshold(address indexed strategy);
```

Comment:
This event doesn't seem very useful, is it missing some fields?

Reply:
Right. It just docs the frequency for analysis on auto-balancing in v1. It'll be more useful in v2 by adding more params, with some restructure to take advantage of v2 design.

Reply:
Why not add the params now?

Reply:
Enough for monitoring. And I believe we will have moved to v2 before this event is triggered :p.